### PR TITLE
[irods/irods#6020] Test hook improvements (4-2-stable)

### PR DESCRIPTION
**Pull request by alanking**
adapted from changes made for #112

### CODE REVIEW

File: irods_consortium_continuous_integration_test_hook.py [hunk 1]
```diff
@@ -13,18 +13,27 @@ def main():
...
-    irods_python_ci_utilities.subprocess_get_output(['sudo', 'su', '-', 'irods', '-c', 'python2 scripts/setup_python_rule_engine_as_only_rule_engine.py'], check_rc=True)
+    if options.do_setup:
+        built_packages_root_directory = options.built_packages_root_directory
+        plugin_dir = irods_python_ci_utilities.append_os_specific_directory(built_packages_root_directory)
+        irods_python_ci_utilities.install_os_packages_from_files([os.path.join(plugin_dir, entry) for entry in os.listdir(plugin_dir)])
+        irods_python_ci_utilities.subprocess_get_output(['sudo', 'su', '-', 'irods', '-c', 'python2 scripts/setup_python_rule_engine_as_only_rule_engine.py'], check_rc=True)
 
     test_output_file = 'log/test_output.log'
+
+    test_option = '--run_s={}'.format(options.test) if options.test else '--run_python_suite'
```
Comment (d-w-moore) [inline irods_consortium_continuous_integration_test_hook.py:28]: Should it be the full `--run-specific_test`, just to make it clearer?

File: irods_consortium_continuous_integration_test_hook.py [hunk 2]
```diff
@@ -13,18 +13,27 @@ def main():
     parser = optparse.OptionParser()
     parser.add_option('--output_root_directory')
     parser.add_option('--built_packages_root_directory')
+    parser.add_option('--test', metavar='dotted name')
+    parser.add_option('--skip-setup', action='store_false', dest='do_setup', default=True)
```
Comment (d-w-moore) [inline irods_consortium_continuous_integration_test_hook.py:17]: Just for my information, what will we use `--skip-setup` for?

Also, I'm thinking `--test` can be a whitespace-delimited list of dotted names, correct?  Wonder if that should be made part of the help text.  (Just a suggestion, it might be more obvious to others....)

Comment (d-w-moore) [inline irods_consortium_continuous_integration_test_hook.py:17]: Just for my information, what will we use `--skip-setup` for?

Also, I'm thinking `--test` can be a whitespace-delimited list of dotted names, correct?  Wonder if that should be made part of the help text.  (Just a suggestion, it might be more obvious to others....)

**Review by d-w-moore — APPROVED** [review]

File: irods_consortium_continuous_integration_test_hook.py (hunk 1 above)
Comment (alanking) [reply irods_consortium_continuous_integration_test_hook.py:28]: Sure thing. This is what the other test hooks use, but I'm fine to add the full thing.

**Comment by alanking** [comment]
@d-w-moore, let me know if you see anything else you'd like changed for this. Thanks!

**Comment by alanking** [comment]
@d-w-moore, let me know if you see anything else you'd like changed for this. Thanks!

**Review by trel — APPROVED** [review]

**Comment by alanking** [comment]
I'm gonna go ahead and merge this to move my stuff along. Any additional changes can come later.
